src/core/mempool.rs

```diff
@@ -1223,6 +1289,14 @@ impl MemPoolDB {
         Ok(rows)
     }
 
+    // Return the results in a very "minimal", or "lazy", representation.
+    // Note: Delay deserialization until we know we want to process this.
+    pub fn get_all_txs_minimal(conn: &DBConn) -> Result<Vec<MemPoolTxMinimalInfo>, db_error> {
```

Comment:
This approach does not scale.  It will appear to work with "small" sample sizes, but the size of the mempool (and thus the size of this dataset) is _O(number-of-addresses)_, which only grows over time.  Also, you're sorting it in-RAM, which should be unnecessary if you just index the fields and structure your queries such that `ORDER BY` statements can use the index(es).

Reply:
i have measurements in the log of how long this takes.

i reiterate that this processed a 100k mint (100k objects in mempool) 

1) it does scale
2) scaling in the size of the mempool is precisely what this was meant to achieve!

i will run a new version of the stress test and upload the full logs so you can see

Reply:
No, @gregorycoppola, you are reading an unbounded amount of data into RAM that _is remotely controlled by attackers_.  Not only does that factually not scale (since RAM is bound), but also it's _dangerous_, and sets us up for failure down the road.

Maybe you can fixed-sized read batches of rows into RAM instead?  Then at least you're bounding the amount of state you're pulling into RAM, as well as bounding the amount of I/O each pass will do.

Reply:
we can sort in ram by fee rate and then take a limited amount into RAM